signal_timeout_block.py

```diff
@@ -51,11 +54,12 @@ def process_group(self, signals, key):
 
         # Lock around the individual group
         with self._jobs_locks[key]:
+            print(key)
             # Cancel any existing timeout jobs, then reschedule them
             self._cancel_timeout_jobs(key)
             for interval in self.intervals:
                 self._schedule_timeout_job(
-                    key, interval.interval, interval.repeatable)
+                    signals[-1:][0], key, interval.interval, interval.repeatable)
```

Comment:
`signals[-1]` I think looks cleaner than `signals[-1:][0]`, is there a reason for the list thing?